php/class-post-type.php

```diff
@@ -822,4 +867,77 @@ public function admin_show_merge_error() {
 		}
 		printf( '<div class="notice notice-error is-dismissible"><p>%s</p></div>', esc_html( $error[ $error_code ] ) );
 	}
+
+	/**
+	 * Prints admin underscore templates.
+	 */
+	public function snapshot_admin_script_template() {
+		global $post;
+		if ( isset( $post->post_type ) && static::SLUG === $post->post_type ) { ?>
+			<script type="text/html" id="tmpl-snapshot-fork-item">
+				<li><a href="{{data.edit_link}}">{{data.post_title}}</a></li>
+			</script>
+		<?php }
+	}
+
+	/**
+	 * Handles snapshot fork ajax request.
+	 */
+	public function handle_snapshot_fork() {
+		$post_type_object = get_post_type_object( self::SLUG );
+		$is_authorized_request = ( ! check_ajax_referer( 'snapshot-fork', 'nonce', false )
+		                           ||
+		                           ! isset( $_POST['ID'] )
+		                           ||
+		                           ! intval( $_POST['ID'] )
+		                           ||
+		                           ! current_user_can( $post_type_object->cap->edit_post, $_POST['ID'] )
```

Comment:
`intval( $_POST['ID'] )` should be passed into the `current_user_can()` check instead of a bare `$_POST['ID']`. As it stands right now, this is passing an unsanitized post value into `current_user_can()`. For example, if `$_POST['ID']` is `'234abc'` then this will run through `intval()` as `234` and pass that check, but then it will get passed in as that unsanitized string to the `current_user_can()`.

You should also make sure that there actually exists a post for the given post and that the post type matches. Looks like you're doing that down below as well, but I think the post's existence and type should be verified before it gets run through the cap check.

You may consider breaking up this condition into multiple `if` statements and then moving `$post_id = intval( $_POST['ID'] );` up so that you can re-use `$post_id`.